[Feat] 게시물 좋아요 기능 구현

ryuneng opened the pull request:
## 📌 작업 내용 (필수)
- 게시물 좋아요 기능을 구현하였습니다.

<br>

## 🌱 반영 브랜치
- feat/#ALL-28-post-like -> dev
- close #66

ryuneng committed:
[Feat] 게시물 좋아요 기능 구현

ryuneng committed:
[Feat] 게시물 좋아요, 공유 관련 외부 API 호출 기능 추가

ryuneng committed:
Merge branch 'dev' of https://github.com/wanted-all-clear/socialhub into feat/#ALL-28-post-like

ryuneng committed:
[Refactor] 존재하지 않는 게시물 예외 처리 추가

ryuneng committed:
[Fix] Pull 받기 전 Push하지 않은 상태의 Commit 내역과 상이한 파일 커밋

ryuneng committed:
[Fix] PostLikeResponse의 좋아요 수 변수명 수정 (likeCount -> likeCnt)

- Post에서 사용하는 변수명과 동일하게 변경하여 일관성을 유지하였습니다.

youhyeoneee commented on src/main/java/com/allclear/socialhub/post/service/PostServiceImpl.java:
```diff
@@ -160,4 +200,35 @@ private Post postCheck(Long postId) {
+        try {
+            restTemplate.postForObject(url, request, String.class);
+        } catch (Exception e) {
+            System.out.println("외부 API 호출에 실패하였습니다." + e.getMessage());
```
print보다 log활용해주시면 더 좋을 것 같습니다!

ryuneng replied:
의견 감사합니다! 반영해서 커밋하겠습니다.

youhyeoneee commented on src/main/java/com/allclear/socialhub/post/service/PostServiceImpl.java:
```diff
@@ -160,4 +200,35 @@ private Post postCheck(Long postId) {
+
+        try {
+            restTemplate.postForObject(url, request, String.class);
+        } catch (Exception e) {
```
Exception이 더 구체적이면 좋을 것 같습니다!

ryuneng replied:
의견 감사합니다! 반영해보겠습니다~

youhyeoneee commented on src/main/java/com/allclear/socialhub/post/service/PostServiceImpl.java:
```diff
@@ -132,6 +139,39 @@ public PostPaging getPosts(Pageable pageable) {
+     *
+     * @param postId 게시물 번호
+     * @param userId 유저 번호
+     */
```
주석에 @return 추가해주시면 감사하겠습니다~!

ryuneng replied:
넵 추가하겠습니다!

youhyeoneee reviewed (commented):
고생하셨습니다!!

bsjin1122 commented:
유현님 코멘트 포함 확인하였습니다~ 고생하셨어요

bsjin1122 commented:
확인하였습니다~

ryuneng committed:
Merge branch 'dev' of https://github.com/wanted-all-clear/socialhub into feat/#ALL-28-post-like

ryuneng committed:
[Refactor] 게시물 좋아요 관련 예외 처리 수정

- 외부 API 호출 예외를 RestClientException으로 구체화하였습니다.
- 외부 API 호출 실패 예외 메시지를 print 대신 log로 기록하도록 변경하였습니다.
